spec/elastic_apm/fields_spec.rb

```diff
@@ -34,12 +35,25 @@ class MyObject
     end
 
     it "knows its fields" do
-      expect(MyObject.fields).to eq(%i[name address])
+      expect(MyObject.schema.keys).to eq(%i[name address online])
+    end
+
+    describe "with default value" do
+      it 'sets value to default' do
+        subject = MyObject.new
+        expect(subject.online).to be true
+      end
+
+      it 'is overridable' do
+        subject = MyObject.new(online: false)
+        expect(subject.online).to be false
+      end
     end
 
     describe "#empty?" do
       it "is when missing all values" do
         subject = MyObject.new
+        subject.online = nil
         expect(subject).to be_empty
       end
```

Comment:
I think we should add a test for when only optional fields have a value. For example:

```ruby
    describe "#empty?" do
      class MyOptionalObject
        include Fields

        field :age, optional: true
        field :language, optional: true
      end

      it "doesn't consider optional fields" do
        subject = MyOptionalObject.new
        subject.age = 11
        subject.language = 'english'
        expect(subject).to be_empty
      end
    end
```
Is that even the behavior we would expect?

Reply:
Thinking about it again, I don't even know what _optionals_ are supposed to signify.